### PR TITLE
Handles general json parse errors and special throttled case.

### LINKED ISSUE #45: Sending too many requests results in obscure error
Hi

If you send too many RTM connection requests, you will eventually receive the following response

`You are sending too many requests. Please relax.`

which due to being plain text and not JSON (response.body doesnt exist), fails with following error 

```
ArgumentError) argument error
        (jsx) src/jsx_decoder.erl:234: :jsx_decoder.value/4
      (exjsx) lib/jsx.ex:15: JSX.decode!/2
      (slack) lib/slack/rtm.ex:8: Slack.Rtm.start/1

```

### PR DESCRIPTION
**Pull request by christhekeele**
Fixes #45.

Took me about 30 requests to connect in a second to trigger, but now it won't let me make more than one a second, which is what the docs claim is the hard cap--so I guess they're pretty forgiving!

### CODE REVIEW

File: lib/slack.ex [hunk 1]
```diff
@@ -95,6 +95,8 @@ defmodule Slack do
             {:error, "Timed out while connecting to the Slack RTM API"}
           {:error, %HTTPoison.Error{reason: :nxdomain}} ->
             {:error, "Could not connect to the Slack RTM API"}
+          {:error, %JSX.DecodeError{string: "You are sending too many requests. Please relax."}} ->
```
Comment (BlakeWilliams) [inline lib/slack.ex]: Thoughts on `string: error` and sending down what Slack gives us?

File: lib/slack/rtm.ex [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+      {:ok, %HTTPoison.Response{body: body} } ->
+        case JSX.decode(body, [{:labels, :atom}]) do
+          {:ok, json}       -> {:ok, json}
+          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body} }
+        end
       {:error, reason} -> {:error, reason}
     end
   end
-end
+end
```
Comment (BlakeWilliams) [inline lib/slack/rtm.ex]: Can you configure your editor to add newlines at the end of files?

File: lib/slack/rtm.ex [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
 
   def start(token) do
     case HTTPoison.get(@url <> token) do
-      {:ok, response} ->
-        json = JSX.decode!(response.body, [{:labels, :atom}])
-        {:ok, json}
+      {:ok, %HTTPoison.Response{body: body} } ->
+        case JSX.decode(body, [{:labels, :atom}]) do
+          {:ok, json}       -> {:ok, json}
+          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body} }
```
Comment (BlakeWilliams) [inline lib/slack/rtm.ex]: Extra space before `}`

File: lib/slack/rtm.ex (hunk 2 above)
Comment (christhekeele) [inline lib/slack/rtm.ex]: Yep! Atom, in this case. Pretty common in vim and Sublime too I think.

**Pull request merged**